### PR TITLE
add throttling error to rescue for retry

### PR DESCRIPTION
**Pull request by mi-wood**
@optimizely/data-services

### CODE REVIEW

File: providers/record.rb [hunk 1]
```diff
@@ -1,5 +1,10 @@
-record_wait_time = 20
-record_wait_tries = 10
+# 'definied?' is for chefspec warnings
+RECORD_WAIT_TIME = 20 unless defined? RECORD_WAIT_TIME
```
Comment (krishnan-chandra) [inline providers/record.rb:2]: Where would this get defined otherwise?

Comment (mi-wood) [inline providers/record.rb:2]: chefspec reloads the same file multiple times :/

Comment (mi-wood) [inline providers/record.rb:2]: I'll make the comment more clear and fix the typo

**Comment by krishnan-chandra** [comment]
LGTM

**Pull request merged**